base/array.jl

```diff
@@ -402,17 +402,13 @@ julia> getindex(Int8, 1, 2, 3)
 """
 function getindex(::Type{T}, vals...) where T
     a = Vector{T}(undef, length(vals))
-    @inbounds for i = 1:length(vals)
-        a[i] = vals[i]
+    afoldl(1, vals...) do i, v
+        @inbounds a[i] = v
+        return i + 1
```

Comment:
What is this line good for? Is this for type stability in the (unconsumed) return type?

Reply:
IIUC, this would be unrolled on julia side thus we avoid union-split in for loop.